### ISSUE TITLE
Replace `number[]` with `Uint8Array`

### ISSUE THREAD

**Issue by JonasKruckenberg** [tauri-apps/tauri#3306]
Our api functions that deal with binary data currently have the type `number[]` which serializes to the correct JSON, but has sub-par DX when used with other JavaScript tools and libraries. 
The attached PR replaces all occurrences of `number[]` with `Uint8Array` the correct `Vec<u8>` equivalent.

This is a breaking change though, as functions no longer accept plain number arrays. Typescript users will get correct errors, but non-ts users will receive `invalid type: map, expected a sequence`

This change will only affect the user facing interface, as Uint8Arrays will be converted into `number[]` under the hood to get  preserve the correct serialization

**Comment by FabianLars** [comment]
tl;dr of my discord messages:
- i don't like that it adds a conversion (for potentially _huge_ arrays) in our most bottlenecked function.
- from what i've seen so far (in tauri-land), users still mostly create normal arrays if created "by hand". This is not really relevant for files and byte bodies tho, which i ignored at first, which is why i **was** against this change.

> but non-ts users will receive invalid type: map, expected a sequence

That's how it _currently_ is. With that pr it would actually work better for non-js users. They won't get an error anymore if they have uint8arrays and because of non-existent typechecking, a number[] array will work too (Array.from(number[]) == number[]).
So for js users you could call this change a fix instead of dx improvement lol

**Comment by JonasKruckenberg** [comment]
> That's how it _currently_ is. With that pr it would actually work better for non-js users. They won't get an error anymore if they have uint8arrays and because of non-existent typechecking, a number[] array will work too (Array.from(number[]) == number[]). So for js users you could call this change a fix instead of dx improvement lol

Ohhh you are right, so this wouldn't be thaat breaking. That is actually awesome!
